src/levanter/models/llama.py

```diff
@@ -288,8 +298,8 @@ def __call__(self, x: NamedArray, mask: Optional[NamedArray], *, key=None) -> Na
         if self.config.upcast_attn:
             q = q.astype(jnp.float32)
             k = k.astype(jnp.float32)
-        k = k.rename({"position": "key_position"})
-        v = v.rename({"position": "key_position"})
+        k = k.rename({"position": "key_position"}).broadcast_axis(self.config.QHeadsPerGroup)
+        v = v.rename({"position": "key_position"}).broadcast_axis(self.config.QHeadsPerGroup)
```

Comment:
i think you can get away without this, but I could be wrong?

Reply:
you're right. in that case, should I define the `QHeadsPerGroup` axis in the attention's `init` function instead of in the config cuz we don't need to reference this axis object other than in the `init`.

Reply:
sure that makes senes

Reply:
I guess the difference is that we'll end up defining this axes using the same name on every layer. Is this considered bad? should the axis be somewhat a "global" thing as it's defined in the config?

Reply:
i think it's fine either way. Up to you what seems cleaner

Reply:
done.